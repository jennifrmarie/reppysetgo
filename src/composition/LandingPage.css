.login {
    text-align: center;
    font-size: 20px;
    color:black;
    margin-top: 100px;
}

.login_new_account {
    margin-top: 90px;
}

.login_button {
    font-size: 20px;
    margin-top: 10px;
    background-color: black;
    color: white;
    padding: 10px;
}