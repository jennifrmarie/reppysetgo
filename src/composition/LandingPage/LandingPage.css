.login {
    text-align: center;
    font-size: 20px;
    color:black;
    margin-top: 100px;
}

.or {
    margin: 20px;
}

.account_button {
    background-color: black;
    color: white;
}


.account_button:hover {
    cursor: pointer;
    background-color: #ad4141fb;;
    color: white;
}

.login_button {
    font-size: 20px;
    margin-top: 10px;
    background-color: black;
    color: white;
    padding: 10px;
}

.login_button:hover {
    cursor: pointer;
    background-color: #ad4141fb;
}