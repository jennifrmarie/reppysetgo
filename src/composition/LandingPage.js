import React, { Component } from 'react'
import config from '../config'
import './LandingPage.css'

export default class LandingPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
        }
        this.handleUsernameInput.bind(this);
        this.handlePasswordInput.bind(this);
    }

    state = { error: null }
    handleUsernameInput(e) {
        this.setState({
            username: e.target.value,
        })        
    }
    handlePasswordInput(e) {
        this.setState({
            password: e.target.value,
        })     
    }
    saveAuthToken(token) {
      window.localStorage.setItem(config.TOKEN_KEY, token)
    }


    handleSubmit = (e) => {
      e.preventDefault()
      const user_name = e.target.user_name.value
      const password = e.target.password.value
      const data = { user_name, password }
      
      fetch(`${config.API_ENDPOINT}api/auth/login`, {
          method: 'post',
          mode: 'no-cors',
          headers: {
              "content-type": "application/json",
          },

          body: JSON.stringify(data)
      })
      .then((res) => {
        if(!res.ok) {
          throw new Error(window.alert('incorrect username/password'))
          
        }
        return res.json();
      })
      .then(data => {
          localStorage.authToken = data.authToken;
          this.props.history.push('/dashboard')
      })
      .catch((error) => {
          console.error('Error')
      })
    }

    handleCreateAccount = (e) => {
      e.preventDefault()
      this.props.history.push('/sign-up')
    }

    render() {
        return (
              <form onSubmit={this.handleSubmit}>
              <div className="logo__login"></div>
                <div className="login__form__credentials">
                <h3 class="login__header">Log In!</h3>
                  <input class="username" type="text" placeholder="guest" name="user_name" id="user_name"/>
                  <input class="password" type="password" placeholder="Test123!" name="password" id="password"/>
                    <div className="login__controls">
                    <input type="checkbox" name="rememberMe" id="rememberMe"/>
                    <label htmlFor="rememberMe">Remember me</label>
                    </div>
                  <button class="login_button">Login</button>
                  <div class="login_new_account">
                  <button className="account_button" onClick={this.handleCreateAccount}>Create a New Account</button>
                  </div>
                </div>
              </form>
        )
    }
}
