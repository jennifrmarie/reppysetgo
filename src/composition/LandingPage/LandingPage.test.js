import React from 'react';
import ReactDOM from 'react-dom';
import LandingPage from './LandingPage';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LandingPage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<LandingPage name="Messages" unread={4}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });