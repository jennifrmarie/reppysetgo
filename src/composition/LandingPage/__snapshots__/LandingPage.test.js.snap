// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="logo__login"
  />
  <div
    className="login__form__credentials"
  >
    <h3
      className="login__header"
    >
      Log In!
    </h3>
    <input
      className="username"
      id="user_name"
      name="user_name"
      placeholder="guest"
      required={true}
      type="text"
    />
    <input
      className="password"
      id="password"
      name="password"
      placeholder="Test123!"
      required={true}
      type="password"
    />
    <div
      className="login__controls"
    >
      <input
        id="rememberMe"
        name="rememberMe"
        type="checkbox"
      />
      <label
        htmlFor="rememberMe"
      >
        Remember me
      </label>
    </div>
    <button
      className="login_button"
    />
    <div
      className="login_new_account"
    >
      <button
        className="account_button"
        onClick={[Function]}
      >
        Create a New Account
      </button>
    </div>
  </div>
</form>
`;
